### PR TITLE
Remove use of Table from 'docker diff'

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: daemon/changes.go [hunk 1]
```diff
@@ -17,21 +18,17 @@ func (daemon *Daemon) ContainerChanges(job *engine.Job) error {
...
 		return err
 	}
 
-	for _, change := range changes {
-		out := &engine.Env{}
-		if err := out.Import(change); err != nil {
-			return err
-		}
-		outs.Add(out)
+	data, err := json.Marshal(changes)
```
Comment (LK4D4) [inline daemon/changes.go]: I think here is better to use `json.NewEncoder` to encode changes right to `job.Stdout`

**Comment by duglin** [comment]
ping @tiborvass @crosbymichael @jfrazelle

File: api/client/diff.go [hunk 2]
```diff
@@ -25,13 +28,14 @@ func (cli *DockerCli) CmdDiff(args ...string) error {
 		return err
 	}
 
-	outs := engine.NewTable("", 0)
-	if _, err := outs.ReadListFrom(body); err != nil {
+	changes := []types.ContainerChange{}
+	if err := json.Unmarshal([]byte(body), &changes); err != nil {
```
Comment (tiborvass) [inline api/client/diff.go]: use `json.NewDecoder(body).Decode(&changes)` instead, you'll need to simply remove the `readBody` wrap.

File: api/types/types.go [hunk 3]
```diff
@@ -35,3 +35,9 @@ type ContainerWaitResponse struct {
 type ContainerCommitResponse struct {
 	ID string `json:"Id"`
 }
+
+// GET /containers/.../changes
```
Comment (tiborvass) [inline api/types/types.go]: I suggest `/containers/{name}/changes`. Note to maintainers: we should probably standardize the comments in this file.

**Comment by duglin** [comment]
@tiborvass @LK4D4 ok updated per your comments

**Comment by tiborvass** [comment]
LGTM, although we'll need to standardize on the comments in types.go.

**Comment by duglin** [comment]
yea I just copied what was in api/server/server.go

File: api/client/diff.go [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
 		return err
 	}
 
-	outs := engine.NewTable("", 0)
-	if _, err := outs.ReadListFrom(body); err != nil {
-		return err
-	}
-	for _, change := range outs.Data {
+	changes := []types.ContainerChange{}
+	json.NewDecoder(rdr).Decode(&changes)
```
Comment (LK4D4) [inline api/client/diff.go]: shouldn't we check errors here?

Comment (duglin) [inline api/client/diff.go]: oops yes we should. Sorry, was thinking/smelling dinner and wasn't thinking straight... :-)  fixed now

**Comment by estesp** [comment]
looks good--I like how this simplifies the code
LGTM

**Pull request merged**